### PR TITLE
Use https in URLs where available

### PR DESCRIPTION
**Pull request by xtqqczze**
## PR Summary

Use https in URLS across the repository where the domain serves content by https.

## PR Checklist

- [x] [PR has a meaningful title](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
    - Use the present tense and imperative mood when describing your changes
- [x] [Summarized changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] [Change is not breaking](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#making-breaking-changes)
- [x] [Make sure all `.h`, `.cpp`, `.cs`, `.ps1` and `.psm1` files have the correct copyright header](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] This PR is ready to merge and is not [Work in Progress](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---work-in-progress).
    - If the PR is work in progress, please add the prefix `WIP:` to the beginning of the title and remove the prefix when the PR is ready.
- **User-facing changes**
    - [x] Not Applicable
- **Testing - New and feature**
    - [x] [Make sure you've added a new test if existing tests do not effectively test the code changed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#before-submitting)
        - [x] [Add `[feature]` if the change is significant or affects feature tests](https://github.com/PowerShell/PowerShell/blob/master/docs/testing-guidelines/testing-guidelines.md#requesting-additional-tests-for-a-pr)

### CODE REVIEW

File: test/common/markdown/package-lock.json [hunk 1]
```diff
@@ -6,7 +6,7 @@
   "dependencies": {
     "ansi-colors": {
       "version": "1.1.0",
-      "resolved": "http://registry.npmjs.org/ansi-colors/-/ansi-colors-1.1.0.tgz",
+      "resolved": "https://registry.npmjs.org/ansi-colors/-/ansi-colors-1.1.0.tgz",
```
Comment (iSazonov) [inline test/common/markdown/package-lock.json]: Do we really need the file package-lock.json?

File: test/powershell/Language/Scripting/ScriptHelp.Tests.ps1 [hunk 2]
```diff
@@ -23,7 +23,7 @@ Describe 'get-help HelpFunc1' -Tags "Feature" {
         It '$x.Synopsis' { $x.Synopsis | Should -BeExactly "A relatively useless function." }
         It '$x.Description' { $x.Description[0].Text | Should -BeExactly "A description`n`n    with indented text and a blank line." }
         It '$x.alertSet.alert' { $x.alertSet.alert[0].Text | Should -BeExactly "This function is mostly harmless." }
-        It '$x.relatedLinks.navigationLink[0].uri' {  $x.relatedLinks.navigationLink[0].uri | Should -BeExactly "http://blogs.msdn.com/powershell" }
+        It '$x.relatedLinks.navigationLink[0].uri' {  $x.relatedLinks.navigationLink[0].uri | Should -BeExactly "https://blogs.msdn.com/powershell" }
```
Comment (iSazonov) [inline test/powershell/Language/Scripting/ScriptHelp.Tests.ps1]: Seems we need revert the change.

File: test/powershell/engine/Cdxml/assets/CimTest/CimTest.cdxml [hunk 3]
```diff
@@ -50,7 +50,7 @@
 
     <!-- Set-CimTest -->
       <Cmdlet>
-        <CmdletMetadata Verb="Set" ConfirmImpact="Medium" HelpUri="http://go.microsoft.com/fwlink/?LinkID=241963" Aliases="scimt"/>
+        <CmdletMetadata Verb="Set" ConfirmImpact="Medium" HelpUri="https://go.microsoft.com/fwlink/?LinkID=241963" Aliases="scimt"/>
```
Comment (iSazonov) [inline test/powershell/engine/Cdxml/assets/CimTest/CimTest.cdxml]: It works?

File: test/powershell/engine/Help/assets/Microsoft.PowerShell.LocalAccounts_8e362604-2c0b-448f-a414-a6a690a644e2_HelpInfo.xml [hunk 4]
```diff
@@ -1,10 +1,10 @@
 ﻿<?xml version="1.0" encoding="utf-8"?>
 <HelpInfo xmlns="http://schemas.microsoft.com/powershell/help/2010/05">
-  <HelpContentURI>http://go.microsoft.com/fwlink/?LinkId=717973</HelpContentURI>
+  <HelpContentURI>https://go.microsoft.com/fwlink/?LinkId=717973</HelpContentURI>
```
Comment (iSazonov) [inline test/powershell/engine/Help/assets/Microsoft.PowerShell.LocalAccounts_8e362604-2c0b-448f-a414-a6a690a644e2_HelpInfo.xml]: It works?
Should we change previous line too?

File: assets/patch-template.wxs [hunk 5]
```diff
@@ -3,7 +3,7 @@
     <Patch 
         AllowRemoval="yes"
         Manufacturer="Microsoft Powershell" 
-        MoreInfoURL="http://github.com/powershell/powershell"
+        MoreInfoURL="https://github.com/powershell/powershell"
```
Comment (iSazonov) [inline assets/patch-template.wxs]: @TravisEz13  Can we change this?

**Comment by iSazonov** [comment]
How can we really check that all the links work?

File: test/powershell/Language/Scripting/ScriptHelp.Tests.ps1 (hunk 2 above)
Comment (xtqqczze) [reply test/powershell/Language/Scripting/ScriptHelp.Tests.ps1]: Test passes because line 52 is also updated.

**Comment by vexx32** [comment]
@iSazonov if I recall correctly a recent PR added link verifications to the `CI-static-analysis` check.

File: test/powershell/engine/Help/assets/Microsoft.PowerShell.LocalAccounts_8e362604-2c0b-448f-a414-a6a690a644e2_HelpInfo.xml (hunk 4 above)
Comment (xtqqczze) [reply test/powershell/engine/Help/assets/Microsoft.PowerShell.LocalAccounts_8e362604-2c0b-448f-a414-a6a690a644e2_HelpInfo.xml]: 302 redirect to http://download.microsoft.com/download/2/3/0/23073AB1-2A89-45F0-8673-FBBA4636AF06/
and then 403 (forbidden)

The previous line refers to a XML namespace not a URL so should be be changed.

File: test/powershell/engine/Cdxml/assets/CimTest/CimTest.cdxml (hunk 3 above)
Comment (xtqqczze) [reply test/powershell/engine/Cdxml/assets/CimTest/CimTest.cdxml]: With missing slash added `http://technet.microsoft.com/library/jj635727(v=wps.620).aspx` responds `301` Moved Permanently to `http://docs.microsoft.com/en-us/powershell/module/smbshare/set-smbshare?view=winserver2012-ps`

**Comment by iSazonov** [comment]
@vexx32 There is only md-file check. 
```
mdspell '**/*.md' '!**/Pester/**/*.md'
```

**Comment by RDIL** [comment]
@iSazonov no you are wrong and @vexx32 is right, I added a link check task but it is only for markdown files.

**Review by RDIL — APPROVED** [review]
Looks good to me, somebody needs to look over the static analysis logs however.

**Comment by RDIL** [comment]
```
2019-01-12T17:16:07.6158436Z __UNELEVATED_TESTS_THE_END__
2019-01-12T17:16:08.2464998Z 1 tests in test/powershell failed
2019-01-12T17:16:08.2465114Z At D:\a\1\s\build.psm1:1348 char:13
2019-01-12T17:16:08.2465163Z +             throw "$($x.'test-results'.failures) tests in $TestArea f ...
2019-01-12T17:16:08.2465227Z +             ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
2019-01-12T17:16:08.2465281Z     + CategoryInfo          : OperationStopped: (1 tests in test/powershell failed:String) [], RuntimeException
2019-01-12T17:16:08.2465330Z     + FullyQualifiedErrorId : 1 tests in test/powershell failed
2019-01-12T17:16:08.2465801Z  
2019-01-12T17:16:08.2465857Z Upload CoreCLR Non-Admin test results
2019-01-12T17:16:08.2465917Z WARNING: Not running in appveyor, skipping upload of test results: D:\a\1\s\TestsResultsNonAdmin.xml
2019-01-12T17:16:08.2466187Z TEST FAILURES
2019-01-12T17:16:08.2466242Z Description: Should match native commands
2019-01-12T17:16:08.2466300Z Name:        Get-Command CI tests.-UseFuzzyMatch tests.Should match native commands
2019-01-12T17:16:08.2466363Z message:
2019-01-12T17:16:08.2466427Z Expected 'PING.EXE' to be found in collection @('gin', 'gin', 'gin', 'pfn', 'dir', 'popd', 'pushd', 'cpi', 'gi', 'iex'), but it was not found.
2019-01-12T17:16:08.2466485Z stack-trace:
2019-01-12T17:16:08.2466561Z at <ScriptBlock>, D:\a\1\s\test\powershell\Modules\Microsoft.PowerShell.Utility\Get-Command.Tests.ps1: line 20
2019-01-12T17:16:08.2466620Z 20:             $cmds.Name | Should -Contain $ping
2019-01-12T17:16:08.2540098Z ##[error]PowerShell exited with code '1'.
```

**Comment by iSazonov** [comment]
Sorry.
@xtqqczze Please rebase to pass CIs.

File: test/common/markdown/package-lock.json (hunk 1 above)
Comment (RDIL) [reply test/common/markdown/package-lock.json]: I think its better to secure links to `npm` to prevent package intercepts

File: assets/license.rtf [hunk 6]
```diff
@@ -1,390 +1,389 @@
-{\rtf1\adeflang1025\ansi\ansicpg1252\uc1\adeff0\deff0\stshfdbch0\stshfloch31506\stshfhich31506\stshfbi31506\deflang1033\deflangfe1033\themelang1033\themelangfe0\themelangcs0{\fonttbl{\f0\fbidi \froman\fcharset0\fprq2{\*\panose 02020603050405020304}Times New Roman;}{\f2\fbidi \fmodern\fcharset0\fprq1{\*\panose 02070309020205020404}Courier New;}
```
Comment (TravisEz13) [inline assets/license.rtf]: Please don't change this file

Comment (TravisEz13) [reply assets/license.rtf]: If something needs changes, please open an issue, with images of what needs to be changed

File: ThirdPartyNotices.txt [hunk 7]
```diff
@@ -3,7 +3,7 @@ THIRD PARTY SOFTWARE NOTICES AND INFORMATION
 Do Not Translate or Localize
 
 This software incorporates material from third parties.
-Microsoft makes certain open source code available at http://3rdpartysource.microsoft.com,
```
Comment (TravisEz13) [inline ThirdPartyNotices.txt]: This is automatically generated, your change will not persist.

Comment (TravisEz13) [reply ThirdPartyNotices.txt]: @dabutvin Is there somewhere else they can make this change?

File: assets/AppImageThirdPartyNotices.txt [hunk 8]
```diff
@@ -6,7 +6,7 @@
 
 
 Copyright (c) 1991-2016 Unicode, Inc. All rights reserved.
-Distributed under the Terms of Use in http://www.unicode.org/copyright.html
+Distributed under the Terms of Use in https://www.unicode.org/copyright.html
```
Comment (TravisEz13) [inline assets/AppImageThirdPartyNotices.txt]: this is supposed to be the third party license as is.  You should make sure this change is also made in the third party license

File: assets/additionalAttributions.txt [hunk 9]
```diff
@@ -62,7 +62,7 @@ All rights reserved.
 
 Licensed under the Apache License, Version 2.0 (the "License");
```
Comment (TravisEz13) [inline assets/additionalAttributions.txt]: The PackageManagement license has changed.  Can you update to this https://github.com/OneGet/oneget/blob/WIP/LICENSE 

Otherwise, revert this change and file an issue, mentioning me.

File: assets/patch-template.wxs (hunk 5 above)
Comment (TravisEz13) [reply assets/patch-template.wxs]: It's fine.

File: ThirdPartyNotices.txt (hunk 7 above)
Comment (dabutvin) [reply ThirdPartyNotices.txt]: 👍  Just pushed an update to the template and atted you on it

File: assets/additionalAttributions.txt (hunk 9 above)
Comment (TravisEz13) [reply assets/additionalAttributions.txt]: @RDIL 	has a PR open #8676

File: test/powershell/engine/Cdxml/assets/CimTest/CimTest.cdxml (hunk 3 above)
Comment (xtqqczze) [reply test/powershell/engine/Cdxml/assets/CimTest/CimTest.cdxml]: Issue created: #8688

File: assets/AppImageThirdPartyNotices.txt (hunk 8 above)
Comment (xtqqczze) [reply assets/AppImageThirdPartyNotices.txt]: Issue created: #8689

File: test/powershell/engine/Help/assets/Microsoft.PowerShell.LocalAccounts_8e362604-2c0b-448f-a414-a6a690a644e2_HelpInfo.xml (hunk 4 above)
Comment (xtqqczze) [reply test/powershell/engine/Help/assets/Microsoft.PowerShell.LocalAccounts_8e362604-2c0b-448f-a414-a6a690a644e2_HelpInfo.xml]: Issue created: #8690

File: assets/license.rtf (hunk 6 above)
Comment (xtqqczze) [reply assets/license.rtf]: Issue created: #8691

Comment (xtqqczze) [reply assets/license.rtf]: Fixed in #8646

File: ThirdPartyNotices.txt (hunk 7 above)
Comment (xtqqczze) [reply ThirdPartyNotices.txt]: @dabutvin Could you provide a link to the update?

File: assets/AppImageThirdPartyNotices.txt (hunk 8 above)
Comment (xtqqczze) [reply assets/AppImageThirdPartyNotices.txt]: Reverted in 01f8f794d9cda64a7a43cd4f389164be760da4c6

File: assets/additionalAttributions.txt (hunk 9 above)
Comment (xtqqczze) [reply assets/additionalAttributions.txt]: Resolved in #8676

**Review by TravisEz13 — APPROVED** [review]

**Comment by TravisEz13** [comment]
Thanks for your contribution
